File: exercises/exercise-1-testing-java/src/main/java/de/emaarco/example/application/port/outbound/bike/BikeSubscriptionProcess.java
package de.emaarco.example.application.port.outbound.bike;

import de.emaarco.example.domain.bike.BikeSubscriptionId;

public interface BikeSubscriptionProcess {
	long startSubscription(BikeSubscriptionId id);
	
	void sendPaymentReceived(BikeSubscriptionId id);
	
	void sendRequestCanceled(BikeSubscriptionId id);
	
	void sendBikeReceived(BikeSubscriptionId id);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/de/emaarco/example/application/port/outbound/bike/BikeSubscriptionProcess.kt
package de.emaarco.example.application.port.outbound.bike

import de.emaarco.example.domain.bike.BikeSubscriptionId

interface BikeSubscriptionProcess {
    fun startSubscription(id: BikeSubscriptionId): Long
    fun sendPaymentReceived(id: BikeSubscriptionId)
    fun sendRequestCanceled(id: BikeSubscriptionId)
    fun sendBikeReceived(id: BikeSubscriptionId)
}
